deployment/common/changeset/mcms_helpers.go

```diff
@@ -0,0 +1,177 @@
+	addrTypeVer, err := env.ExistingAddresses.AddressesForChain(chainSelector)
+	if err != nil {
+		return nil, fmt.Errorf("error getting addresses for chain: %w", err)
+	}
+	var timelock *owner_helpers.RBACTimelock
+	var callProxy *owner_helpers.CallProxy
+	for addr, tv := range addrTypeVer {
+		if tv.Type == types.RBACTimelock {
+			if timelock != nil {
+				return nil, fmt.Errorf("multiple timelocks found on chain %d", chainSelector)
+			}
+			var err error
+			timelock, err = owner_helpers.NewRBACTimelock(common.HexToAddress(addr), env.Chains[chainSelector].Client)
+			if err != nil {
+				return nil, fmt.Errorf("error creating timelock: %w", err)
+			}
+		}
+		if tv.Type == types.CallProxy {
+			if callProxy != nil {
+				return nil, fmt.Errorf("multiple call proxies found on chain %d", chainSelector)
+			}
+			var err error
+			callProxy, err = owner_helpers.NewCallProxy(common.HexToAddress(addr), env.Chains[chainSelector].Client)
+			if err != nil {
+				return nil, fmt.Errorf("error creating call proxy: %w", err)
+			}
+		}
+	}
+	if timelock == nil || callProxy == nil {
+		return nil, fmt.Errorf("missing timelock (%T) or call proxy(%T) on chain %d", timelock == nil, callProxy == nil, chainSelector)
+	}
+	return &TimelockExecutionContracts{
+		Timelock:  timelock,
+		CallProxy: callProxy,
+	}, nil
+}
+
+type RunTimelockExecutorConfig struct {
+	Executor          *mcms.Executor
+	TimelockContracts *TimelockExecutionContracts
+	ChainSelector     uint64
+	// BlockStart is optional. It filter the timelock scheduled events.
+	// If not provided, the executor assumes that the operations have not been executed yet
+	// executes all the operations for the given chain.
+	BlockStart *uint64
+	BlockEnd   *uint64
+}
+
+func (cfg RunTimelockExecutorConfig) Validate() error {
+	if cfg.Executor == nil {
+		return fmt.Errorf("executor is nil")
+	}
+	if cfg.TimelockContracts == nil {
+		return fmt.Errorf("timelock contracts is nil")
+	}
+	if cfg.ChainSelector == 0 {
+		return fmt.Errorf("chain selector is 0")
+	}
+	if cfg.BlockStart != nil && cfg.BlockEnd == nil {
+		if *cfg.BlockStart > *cfg.BlockEnd {
+			return fmt.Errorf("block start is greater than block end")
+		}
+	}
+	if cfg.BlockStart == nil && cfg.BlockEnd != nil {
+		return fmt.Errorf("block start must not be nil when block end is not nil")
+	}
+
+	if len(cfg.Executor.Operations[mcms.ChainIdentifier(cfg.ChainSelector)]) == 0 {
+		return fmt.Errorf("no operations for chain %d", cfg.ChainSelector)
+	}
+	return nil
+}
+
+// RunTimelockExecutor runs the scheduled operations for the given chain.
+// If the block start is not provided, it assumes that the operations have not been scheduled yet
+// and executes all the operations for the given chain.
+// It is an error if there are no operations for the given chain.
+func RunTimelockExecutor(env deployment.Environment, cfg RunTimelockExecutorConfig) error {
```

Comment:
Sigh. It makes me nervous having this all accessible to changesets, as I want to avoid changesets themselves from ever executing this logic. But I guess this is fine for now - I agree, it probably ought to go into MCMS library though. We can move it there later.

Reply:
do either of these make it better:
- in a real env (with non zero timelock) it doesn't matter b/c the duration past while in the timelock
- moving pkgs

Reply:
I think both help, yeah.